src/store/user.js

```diff
@@ -0,0 +1,156 @@
+import { userService } from '../services';
+import router from '../router';
+
+const token = localStorage.getItem('token');
+
+const namespaced = true;
+
+const state = token
+  ? { status: { loggedIn: true }, user: { token } }
+  : { status: {}, user: null };
+
+/* Actions */
+const login = ({ dispatch, commit }, { email, password, path }) => {
+  commit('loginRequest', { email });
+
+  userService.login(email, password)
+    .then(
+      (loginToken) => {
+        commit('loginSuccess', loginToken);
+        router.replace(path || '/conversations');
+      },
+      (error) => {
+        commit('loginFailure', error);
+        dispatch('alert/error', error, { root: true });
+      },
+    )
+    .catch((error) => {
+      commit('loginFailure', error);
+      dispatch('alert/error', error, { root: true });
+    });
+};
+
+const logout = ({ commit }) => {
+  userService.logout(() => router.push('/'));
```

Comment:
`router.replace('/')` only actually replaces the current entry, not the entire history, so it doesn't really matter too much in this case because they'll just be able to go back to the route entry before this one. Also we do checks for the other endpoints for the token, so when they go back, they'll be redirected to `/` anyway but at least the route they expect to go to will be used to redirect them after they login, instead of the route before that route in history.

Reply:
Kk, sounds goos